smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptJmesPathVisitor.java

```diff
@@ -0,0 +1,315 @@
+
+        expression.arguments.forEach((JmespathExpression argExpression) -> {
+            argExpression.accept(this);
+            switch (expression.getName()) {
+                case "length":
+                    this.executionContext = this.executionContext + ".length";
+                    break;
+                case "contains":
+                    executionContexts.add(this.executionContext);
+                    break;
+                default:
+                    throw new CodegenException("TypeScriptJmesPath visitor has not implemented function: "
+                            + expression.getName());
+            }
+        });
+
+        if (expression.getName().equals("contains")) {
+            executionContext = String.join(".includes(", executionContexts) + ")";
+        }
+        return null;
+    }
+
+    @Override
+    public Void visitField(FieldExpression expression) {
+        this.executionContext += ".";
+        this.executionContext += expression.getName();
+        return null;
+    }
+
+    @Override
+    public Void visitIndex(IndexExpression expression) {
+        if (expression.getIndex() >= 0) {
+            executionContext += ("[" + expression.getIndex() + "]");
+        } else {
+            executionContext += "[" + executionContext + ".length";
+            executionContext += " - " + Math.abs(expression.getIndex()) + "]";
+        }
+        return null;
+    }
+
+    @Override
+    public Void visitLiteral(LiteralExpression expression) {
+        String expressionValue = expression.getValue().toString();
+        if (TypeScriptJmesPathVisitor.isNumeric(expressionValue)
+            || expressionValue.equals("false")
+            || expressionValue.equals("true")) {
+            this.executionContext = expressionValue;
+            return null;
+        }
+
+        this.executionContext = "'" + expressionValue + "'";
+        return null;
+    }
+
+    @Override
+    public Void visitMultiSelectList(MultiSelectListExpression expression) {
+        ArrayList<String> evaluators = new ArrayList<String>();
+
+        String executionContextInital = this.executionContext;
+
+        expression.getExpressions().forEach((JmespathExpression exp) -> {
+            exp.accept(this);
+            evaluators.add(this.executionContext);
+            this.executionContext = executionContextInital;
+        });
+
+        String resultScope = makeNewScope("result_");
+        writer.write("let $L = [];", resultScope);
+        for (String evaluator : evaluators) {
+            writer.write("$L.push($L);", resultScope, evaluator);
+        }
+        writer.write("$L = $L;", this.executionContext, resultScope);
+
+        return null;
+    }
+
+    @Override
+    public Void visitMultiSelectHash(MultiSelectHashExpression expression) {
+        throw new CodegenException("TypeScriptJmesPath visitor not implemented MultiSelectHashExpression");
+    }
+
+    @Override
+    public Void visitAnd(AndExpression expression) {
```

Comment:
Are we really not already using and, or, and not in AWS waiters? These seem like table stakes.

Reply:
implemented